src/intersections/segments.jl

```diff
@@ -167,3 +167,86 @@ function intersectcolinear(s1::Segment{Dim,T}, s2::Segment{Dim,T}) where {Dim,T}
     NoIntersection()
   end
 end
+
+# intersection of Segment and Triangle
+function intersecttype(s::Segment{3,T}, t::Triangle{3,T}) where {T}
+  # get the triangle's and segment's vertices
+  s_v = s.vertices
+  t_v = t.vertices
+
+  s₁ = s_v[1]
+  s₂ = s_v[2]
+
+  # simple checks to see if the triangle overlaps with the bounding box of the segment
+  # this check is much faster than calculating an intersection
+  if any(i -> all(j -> (s₁.coords[i] > t_v[j].coords[i]) && (s₂.coords[i] > t_v[j].coords[i]), 1:3), 1:3)
```

Comment:
Any chance this algorithm can be agnostic to embedding dimension? It only works for triangles embedded in 3D space or also works for 2D space?

Again, the same comments apply about using public API instead of `s.coords`

Reply:
This part here is certainly agnostic to embedding dimension, I could split out this into a separate function perhaps?

Reply:
No worries, I was just wondering if the entire algorithm was agnostic to dimension.